sklearn/inspection/_partial_dependence.py

```diff
@@ -954,3 +1111,470 @@ def plot(self, ax=None, n_cols=3, line_kw=None, contour_kw=None):
+
+    Parameters
+    ----------
+    estimator : BaseEstimator
+        A fitted estimator object implementing :term:`predict`,
+        :term:`predict_proba`, or :term:`decision_function`.
+        Multioutput-multiclass classifiers are not supported.
+
+    X : {array-like or dataframe} of shape (n_samples, n_features)
+        ``X`` is used both to generate a grid of values for the
+        ``features``, and to compute the predictions.
+
+    features : array-like of {int, str}
+        The feature (e.g. `[0]`) or pair of interacting features
+        (e.g. `[(0, 1)]`) for which the ICE should be computed.
+
+    response_method : 'auto', 'predict_proba' or 'decision_function', \
+            optional (default='auto')
+        Specifies whether to use :term:`predict_proba` or
+        :term:`decision_function` as the target response. For regressors
+        this parameter is ignored and the response is always the output of
+        :term:`predict`. By default, :term:`predict_proba` is tried first
+        and we revert to :term:`decision_function` if it doesn't exist.
+
+    percentiles : tuple of float, optional (default=(0.05, 0.95))
+        The lower and upper percentile used to create the extreme values
+        for the grid. Must be in [0, 1].
+
+    grid_resolution : int, optional (default=100)
+        The number of equally spaced points on the grid, for each target
+        feature.
+
+    fixed_start_point : bool, optional (default=True)
+        Whether to use a fixes starting point for all the ICE curves.
+
+    kwargs : dict
+        Ignored.
+
+    Returns
+    -------
+    predictions : ndarray, \
+            shape (n_outputs, n_instances, len(values[0]), len(values[1]),...)
+        The predictions for all the points in the grid, for all
+        samples in X. ``n_outputs`` corresponds to the number of classes in
+        a multi-class setting, or to the number of tasks for multi-output
+        regression. For classical regression and binary classification
+        ``n_outputs==1``. ``n_values_feature_j`` corresponds to the size
+        ``values[j]``.
+
+    values : seq of 1d ndarrays
+        The values with which the grid has been created. The generated grid
+        is a cartesian product of the arrays in ``values``. ``len(values) ==
+        len(features)``. The size of each array ``values[j]`` is either
+        ``grid_resolution``, or the number of unique values in ``X[:, j]``,
+        whichever is smaller.
+
+    Examples
+    --------
+    >>> X = [[0, 0, 2], [1, 0, 0]]
+    >>> y = [0, 1]
+    >>> from sklearn.ensemble import GradientBoostingClassifier
+    >>> gb = GradientBoostingClassifier(random_state=0).fit(X, y)
+    >>> individual_conditional_expectation(gb, features=[0], X=X,
+    ...       percentiles=(0, 1), grid_resolution=2) # doctest: +SKIP
+    (array([[[2.19017313e-05, 5.16319395e-01],
+        [4.83680605e-01, 9.99978098e-01]]]), [array([0., 1.])])
```

Comment:
We need to add the reference to the original paper (which you earlier added in your example).

Reply:
Added in https://github.com/scikit-learn/scikit-learn/pull/16164/commits/62d112b230c3d66e922bf5c95d4c80ef126e4640